tests/test_sequences_and_iterators.cpp

```diff
@@ -93,6 +117,23 @@ TEST_SUBMODULE(sequences_and_iterators, m) {
         })
         ;
 
+    py::class_<Thing>(m, "Thing")
+        .def(py::init<>())
+        .def("__getitem__",
+                [](Thing &t, py::slice slice) -> py::list {
+                    return get_item_list(t, slice);
+                }
+        )
+        .def("reverse",
+                [](Thing &t) -> py::list {
+                    py::slice ordinary_indices(0, static_cast<ssize_t>(t.size()), 1);
+                    py::slice reversed_slice(py::none(), py::none(), py::int_(-1));
+                    py::list reversed = get_item_list(t, reversed_slice);
```

Comment:
Needs explicit testing for `std::optional<>`. (In my suggested change, I'm either using it wrong or there's a minor bug.)